src/test/groovy/org/gradle/android/JdkImageWorkaroundTest.groovy

```diff
@@ -43,14 +43,10 @@ class JdkImageWorkaroundTest extends AbstractTest {
 
         then:
         buildResult.task(':app:compileDebugJavaWithJavac').outcome == TaskOutcome.SUCCESS
-        buildResult.task(':app:compileReleaseJavaWithJavac').outcome == TaskOutcome.SUCCESS
```

Comment:
a side-effect of removing all the complexity of ksp/kapt in the `SimpleAndroIApp` code.
All the `JdkImageWorkaroundTest` disable the Kotlin plugin, in this scenario where a simple project is not using Kotlin/Kapt/Ksp, since AGP 8.0, the build variants generate the same Build cache key for the JavaCompile task.